corehq/apps/cleanup/dbaccessors.py

```diff
@@ -0,0 +1,68 @@
+from collections import defaultdict
+
+from django.db import connections
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.es import AppES, CaseES, CaseSearchES, FormES, GroupES, UserES
+from corehq.apps.userreports.util import (
+    LEGACY_UCR_TABLE_PREFIX,
+    UCR_TABLE_PREFIX,
+    get_domain_for_ucr_table_name,
+)
+from corehq.form_processor.models import CommCareCase, XFormInstance
+from corehq.sql_db.connections import UCR_ENGINE_ID, ConnectionManager
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+
+
+def find_sql_forms_for_deleted_domains():
+    form_count_by_deleted_domain = {}
+    for domain in Domain.get_deleted_domain_names():
+        form_count = 0
+        for db_name in get_db_aliases_for_partitioned_query():
+            form_count += XFormInstance.objects.using(db_name).filter(domain=domain).count()
+        if form_count:
+            form_count_by_deleted_domain[domain] = form_count
+
+    return form_count_by_deleted_domain
+
+
+def find_sql_cases_for_deleted_domains():
+    case_count_by_deleted_domain = {}
+    for domain in Domain.get_deleted_domain_names():
+        case_count = 0
+        for db_name in get_db_aliases_for_partitioned_query():
+            case_count += CommCareCase.objects.using(db_name).filter(domain=domain).count()
+        if case_count:
+            case_count_by_deleted_domain[domain] = case_count
+
+    return case_count_by_deleted_domain
```

Comment:
There is some duplication that seems like it could be eliminated here. What do you think of refactoring these two functions into a single function with a `model` parameter that would accept `CommCareCase` or `XFormInstance`?

The refactor could implement the combined function as a module-private function, and still retain these same functions that simply call the private function, passing their respective `model` parameter.

Reply:
Yep makes sense. See c2a36775ab79196dd5c420db416657eabfcba47f